plugins/performance-lab/tests/includes/site-health/audit-enqueued-assets/test-audit-enqueued-assets-helper.php

```diff
@@ -6,105 +6,615 @@
+	 * Tests perflab_aea_enqueued_blocking_assets_test
+	 *
+	 * @covers ::perflab_aea_enqueued_blocking_assets_test
+	 * @covers ::perflab_aea_generate_blocking_assets_table
+	 * @covers ::perflab_aea_enqueued_blocking_scripts
+	 * @covers ::perflab_aea_enqueued_blocking_styles
 	 */
-	public function test_perflab_aea_get_total_enqueued_scripts(): void {
-		$total_enqueued_styles = perflab_aea_get_total_enqueued_styles();
-		$this->assertFalse( $total_enqueued_styles );
+	public function test_perflab_aea_enqueued_blocking_assets_test_but_one_style_is_404(): void {
+		Audit_Assets_Mock_Assets::mock_assets( 'scripts', 2 );
+		Audit_Assets_Mock_Assets::mock_assets( 'styles', 3, 1 );
+		Audit_Assets_Mock_Assets::mock_requests();
+		$test = perflab_aea_enqueued_blocking_assets_test();
 
-		Audit_Assets_Transients_Set::set_script_transient_with_data( 5 );
-		$total_enqueued_scripts = perflab_aea_get_total_enqueued_scripts();
-		$this->assertIsInt( $total_enqueued_scripts );
-		$this->assertEquals( 5, $total_enqueued_scripts );
+		$this->assertSameSets(
+			array( 'label', 'status', 'badge', 'description', 'actions', 'test' ),
+			array_keys( $test )
+		);
+		$this->assertSame( 'recommended', $test['status'] );
+
+		$this->assertStringContainsString( 'N/A', $test['description'] );
+		$this->assertStringContainsString( 'Not found', $test['description'] );
 	}
 
 	/**
-	 * Tests perflab_aea_get_total_size_bytes_enqueued_scripts().
+	 * Tests perflab_aea_enqueued_ajax_blocking_assets_test
+	 *
+	 * @covers ::perflab_aea_enqueued_ajax_blocking_assets_test
 	 */
-	public function test_perflab_aea_get_total_size_bytes_enqueued_scripts(): void {
-		$size_enqueued_scripts = perflab_aea_get_total_size_bytes_enqueued_scripts();
-		$this->assertFalse( $size_enqueued_scripts );
+	public function test_perflab_aea_enqueued_ajax_blocking_assets_test_unauthenticated_without_nonce(): void {
+		$this->add_filter_to_mock_front_page_loopback_request();
+		$this->expectException( WPAjaxDieStopException::class );
+		$this->_handleAjax( 'health-check-enqueued-blocking-assets-test' );
+		$response = json_decode( $this->_last_response, true );
+		$this->assertArrayHasKey( 'success', $response );
+		$this->assertFalse( $response['success'] );
+	}
 
-		Audit_Assets_Transients_Set::set_script_transient_with_data( 5 );
-		$total_enqueued_scripts = perflab_aea_get_total_size_bytes_enqueued_scripts();
-		$this->assertEquals( 5000, $total_enqueued_scripts );
+	/**
+	 * Tests perflab_aea_enqueued_ajax_blocking_assets_test
+	 *
+	 * @covers ::perflab_aea_enqueued_ajax_blocking_assets_test
+	 */
+	public function test_perflab_aea_enqueued_ajax_blocking_assets_test_unauthenticated_with_nonce(): void {
+		$this->add_filter_to_mock_front_page_loopback_request();
+		$_GET['_wpnonce'] = wp_create_nonce( 'health-check-site-status' );
+		$this->expectException( WPAjaxDieContinueException::class );
+		$this->_handleAjax( 'health-check-enqueued-blocking-assets-test' );
```

Comment:
See https://github.com/WordPress/performance/pull/2059/files#r2281011145.

```suggestion
		try {
			$this->_handleAjax( 'health-check-enqueued-blocking-assets-test' );
		} catch ( WPAjaxDieContinueException $e ) { // phpcs:ignore Generic.CodeAnalysis.EmptyStatement.DetectedCatch
			// Expected exception for unauthenticated access.
		}
```